Sources/Algorithms/PartialSort.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+
+private final class Heap<T> {
```

Comment:
You've declared a class here, but it looks like all the operations are on just the `elements` array. Would this work as a series of array or `MutableCollection` extensions instead of as a type?

Reply:
Yeah, I liked having a separate type for it but we wouldn't be able to get rid of the escaping attribute if we didn't make this all happen inside the method anyway.